docs/DeveloperGuide.md

```diff
@@ -720,7 +832,88 @@ testers are expected to do more *exploratory* testing.
 
 [trackAScholarImport test files](https://github.com/AY2223S1-CS2103T-W10-3/tp/tree/master/src/test/data/JsonImportCommandTest)
 
+--------------------------------------------------------------------------------------------------------------------
+
+### Pinning an applicant
+
+1. Pinning an existing applicant while all applicants are being shown.
+
+    1. Prerequisites: List all applicants using the `list` command. Multiple applicants in the list. 
 
+    1. Test case: `pin 1`<br>
+       Expected: First applicant in the list is pinned. Details of the pinned applicant shown in the status message. 
+       Pinned Applicant will be shown on the right list panel. Timestamp in the status bar is updated.
+
+    1. Test case: `pin 0`<br>
+       Expected: No applicant is pinned. Error details shown in the status message. Status bar remains the same.
+
+    1. Other incorrect pin commands to try: `pin`, `pin x`, `...` (where x is larger than the list size)<br>
```

Comment:
Perhaps consider adding the test case where the user attempts to pin an already pinned applicant?